### PR TITLE
Fixes BaseBuilder getWhere() bug

### LINKED ISSUE #2143: Query Builder getWhere Crash
**Describe the bug**
When using the Query Builder, the getWhere method crashes if no offset is given with "Argument 2 passed to CodeIgniter\\Database\\BaseBuilder::limit() must be of the type int, null given"

The default offset should be zero, and if an int is entered, it would use that.

**CodeIgniter 4 version**
BETA4

**Affected module(s)**
/vendor/codeigniter4/framework/system/Database/BaseBuilder.php line 1612

**Expected behavior, and steps to reproduce if appropriate**
->getWhere([
	'email' => $email,
], 1);

will crash

->getWhere([
	'email' => $email,
], 1, 0);

will run as expected.

Suggest changing line 1603 of /vendor/codeigniter4/framework/system/Database/BaseBuilder.php

from

public function getWhere($where = null, int $limit = null, int $offset = null)

to

public function getWhere($where = null, int $limit = null, int $offset = 0)

**Context**
 - PHP version 7.3.8

### PR DESCRIPTION
**Pull request by michalsn**
**Description**
Fixes #2143 - `getWhere()` bug with offset.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by jim-parry** [comment]
You have added returnSQL and reset parameters to the getWhere() signature.
These don't seem related to the default values bug.
Am I missing something?

**Comment by michalsn** [comment]
1. `returnSQL` - without it this method is not testable.
2. `reset` - I feel it should be consistent with the `get()` method - it just makes more sense to me... so I don't see a reason not to allow reset in this method too.

**Comment by jim-parry** [comment]
@lonnieezell @MGatner Makes sense. Are you good with this?

**Comment by MGatner** [comment]
It looks good to me, really nice tests. It's unfortunate how all the database & builder methods keep growing in number of parameters but I think @michalsn is right that the current implementation needs those. Maybe down the road some way of globally intercepting DB calls to return SQL could replace some of the individual versions?

**Comment by jim-parry** [comment]
If we are adding two parameters to a method signature, shouldn't we be updating the user guide too?

**Comment by MGatner** [comment]
Yes, updates to [user_guide_src/source/database/query_builder.rst](user_guide_src/source/database/query_builder.rst). If @michalsn isn't around I could pass these tonight as a separate PR.

File: user_guide_src/source/database/query_builder.rst [hunk 1]
```diff
@@ -1084,11 +1084,13 @@ Class Reference
...
 		called Query Builder methods.
 
-	.. php:method:: getWhere([$where = NULL[, $limit = NULL[, $offset = NULL]]])
+	.. php:method:: getWhere([$where = NULL[, $limit = NULL[, $offset = NULL[, $returnSQL = FALSE[, $reset = TRUE]]]]])
 
 		:param	string	$where: The WHERE clause
 		:param	int	$limit: The LIMIT clause
 		:param	int	$offset: The OFFSET clause
+		:param 	bool $returnSQL: If true, returns the generate SQL, otherwise executes the query.
+	    :param 	bool $reset: Are we want to clear query builder values?
```
Comment (MGatner) [inline user_guide_src/source/database/query_builder.rst]: Thanks! Small items:
* Indent looks to be 1 tab short
* Should read "**Do** we want to clear..."

Comment (michalsn) [reply user_guide_src/source/database/query_builder.rst]: Done.

**Comment by MGatner** [comment]
Thanks so much @michalsn, looks great.

**Pull request merged**